applications/nrf_desktop/doc/power_manager.rst

```diff
@@ -1,7 +1,132 @@
+
+With ``CONFIG_DESKTOP_POWER_MANAGER_TIMEOUT`` configuration option you can set time after which the application will go to low power mode.
+The ``CONFIG_DESKTOP_POWER_MANAGER_STAY_ON`` let system stay idle also when there are no active connections.
+The ``CONFIG_DESKTOP_POWER_MANAGER_ERROR_TIMEOUT`` set the time after which the device is turned off upon an internal error.
+
+
+Implementation details
+**********************
+
+The :ref:`nrf_desktop_power_manager` is started when the :ref:`nrf_desktop_main` module is ready (reported using ``module_state_event``).
+
+The :ref:`nrf_desktop_power_manager` takes control of the overall system power state.
+Except when entering a deep sleep mode the system will stay in ``SYS_POWER_STATE_ACTIVE`` state.
+When entering the deep sleep mode the module will set system to ``SYS_POWER_STATE_DEEP_SLEEP_1`` state.
+
+
+Power down
+==========
+
+If the device is not connected to the USB, the module counts time since the last user interaction (last HID report sent from the device).
+If the device is connected to the USB, the module remains idle.
+
+On timeout :ref:`nrf_desktop_power_manager` sets the application to the low power state.
+When switching to the low power state :ref:`nrf_desktop_power_manager` will send a `power_down_event``.
+Other application modules react to this event to change their configuration to low power (e.g. turn off LEDs).
+Only after all modules confirmed entering the low power state (by not consuming the ``power_down_event``) the :ref:`nrf_desktop_power_manager` will sets application state to low power.
+
+The next step depends on the BLE connection state.
+ * If the device is connected to the BLE, the module will perform no more actions and the system will remain in an idle state.
+ * If the device is not connected, the module will switch the system into the deep sleep mode.
+   When entering the deep sleep mode, Zephyr suspends all devices and CPU is set to OFF state.
+   To exit deep sleep mode CPU needs an external event.
+
+If disconnection happens while device is in the low power mode, the :ref:`nrf_desktop_power_manager` will set system to the deep sleep mode as described above.
+
+The application can be configured to keep system in the idle state, instead of entering the deep sleep mode, when there are no active connections.
+To select this behavior use ``CONFIG_DESKTOP_POWER_MANAGER_STAY_ON`` configuration option.
+
+Each module must report entering the low power state by sending the appropriate module state event, when it has done so.
+
+
+Postponed power down
+--------------------
+
+When :ref:`nrf_desktop_power_manager` detects that the application should enter the low power state it sends the ``power_down_event``.
+It is possible that some module will not be ready to switch to the lower power state at that time.
+In such case this module should consume the ``power_down_event`` and change its internal state so it enters the low power state when ready.
```

Comment:
```suggestion
In such case, the module that is not yet ready should consume ``power_down_event`` and change its internal state, so that it enters the low power state when ready.
```
Is there a less figurative way of saying "to consume event"? "complete"? "accept"?